File: Data Structures & Algorithms/two-integer-sum-ii/submission-7.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        i = 0
        j = len(numbers) - 1

        while i < j:
            actual_sum = numbers[i] + numbers[j]
            if actual_sum == target:
                return [i + 1, j + 1]
            elif actual_sum > target:
                j -= 1
            else:
                i += 1

        

File: Data Structures & Algorithms/search-2d-matrix/submission-7.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int rows = matrix.length;
        int columns = matrix[0].length;

        int i = 0;
        int j = (rows * columns) - 1;

        while(i <= j) {
            int mid = (i + j) / 2;
            int midValue = matrix[mid / columns][mid % columns];

            if(midValue == target) {
                return true;
            } else if(midValue > target) {
                j = mid - 1;
            } else {
                i = mid + 1;
            }
        }
        return false;
    }
}
